cmd/snap-update-ns/expand.go

```diff
@@ -56,27 +56,29 @@ func expandXdgRuntimeDir(profile *osutil.MountProfile, uid int) {
 }
 
 // expandHomeDir expands the $HOME variable in the given mount profile for entries
-// of mount kind "ensure-dir". It returns and error if expansion is required but the
-// given homeError indicates that given home cannot be used for expansion.
-func expandHomeDir(profile *osutil.MountProfile, home string, homeError error) error {
+// of mount kind "ensure-dir". It returns and error if expansion is required but home
+// err indicates that path should not be used for expansion.
+func expandHomeDir(profile *osutil.MountProfile, home func() (path string, err error)) error {
 	const envVar = "$HOME"
+	path, err := home()
```

Comment:
homeErr was probably a better name for an error not checked immediately

Reply:
fixed